newswires/app/controllers/QueryController.scala

```diff
@@ -17,21 +24,34 @@ class QueryController(
 ) extends BaseController
     with Logging
     with AppAuthActions {
+  private def paramToList[T](
+      request: Request[T],
+      paramName: String
+  ): List[String] =
+    request.getQueryString(paramName).map(_.split(",").toList).getOrElse(Nil)
 
   def query(
       maybeFreeTextQuery: Option[String],
       maybeKeywords: Option[String],
       suppliers: List[String],
       maybeBeforeId: Option[Int],
       maybeSinceId: Option[Int]
-  ): Action[AnyContent] = apiAuthAction {
+  ): Action[AnyContent] = apiAuthAction { request: UserRequest[AnyContent] =>
+    val queryParams = SearchParams(
+      text = maybeFreeTextQuery,
+      keywordIncl = maybeKeywords.map(_.split(",").toList).getOrElse(Nil),
+      keywordExcl = paramToList(request, "keywordsExcl"),
+      suppliersIncl = suppliers,
+      suppliersExcl =
```

Comment:
is there a reason why we're sometimes using the `paramsToList` method and sometimes not, here?

Reply:
they're subtly different, bun incompatible:
currently `supplier` and `supplierExcl` are repeated query params, rather than single params with comma-separated values. the former is extracted by `request.queryString.get(X)`, the latter by `request.getQueryString(X)` (which is what is in `paramToList`)

keywords is extracted by the mapping in the routes file, rather than directly from the request object here... we should also standardise on the location for that - putting them in the routes file is nice because it can then serve as documentation not only of routes but available parameters, but I'm a little nervous because it's a lot of params of the same type, so it'd be easy to accidentally bind a parameter to the wrong name, which would be painful to debug, so I'm putting that decision off too 😅